lib/pip.js

```diff
@@ -211,36 +417,56 @@ function installAllRequirements() {
         if (!get(f, 'module')) {
           set(f, ['module'], '.');
         }
+        // If we didn't already process a module (functions can re-use modules)
         if (!doneModules.includes(f.module)) {
-          installRequirements(
-            path.join(f.module, this.options.fileName),
-            path.join('.serverless', f.module),
-            this.serverless,
+          const reqsInstalledAt = installRequirementsIfNeeded(
             this.servicePath,
-            this.options
+            f.module,
+            this.options,
+            f,
+            this.serverless
           );
-          if (f.vendor) {
-            // copy vendor libraries to requirements folder
-            copyVendors(
-              f.vendor,
-              path.join('.serverless', f.module),
-              this.serverless
-            );
+          // Add modulePath into .serverless for each module so it's easier for injecting and for users to see where reqs are
+          let modulePath = path.join(
+            this.servicePath,
+            '.serverless',
+            `${f.module}`,
+            'requirements'
+          );
+          if (
+            reqsInstalledAt &&
+            !fse.existsSync(modulePath) &&
+            reqsInstalledAt != modulePath
+          ) {
+            if (this.options.staticCache) {
```

Comment:
Found!

```
this.options.useStaticCache
```

Reply:
See my comment below, I am not quite sure your problem, I tried your config on gitlab and it worked perfectly, using caching perfectly.

Reply:
@AndrewFarley The problem is when you package individually, the code in my gitlab does not do that, that's why it works. The code is wrong here, you typed `staticCache` instead of `useStaticCache`.

Reply:
@felipe-augusto Ahh yeah that was the old variable name... thanks for catching that!  I searched the codebase there were no other instances.  I just pushed a fix, try it now?  :)

Reply:
@AndrewFarley Yes, I can confirm that is actually working! Thanks!

Reply:
Great thanks for the feedback.  Hope it saves you tons of time on your builds like it does us.  :)

Reply:
Of course it does, this needs to be merged ASAP. My build went from 22 minutes to 4. Really awesome work! 🥇